lib/vmdb/config.rb

```diff
@@ -57,8 +57,8 @@ def save(resource = MiqServer.my_server)
     end
 
     # NOTE: Used by Configuration -> Advanced
-    def self.get_file
-      Vmdb::Settings.encrypt_passwords!(::Settings.to_hash).to_yaml
+    def self.get_file(resource = MiqServer.my_server)
+      Vmdb::Settings.encrypt_passwords!(resource.settings_for_resource).to_hash.to_yaml
```

Comment:
@h-kataria You have to put the `.to_hash` inside of the encrypt_passwords, otherwise it modifies the passwords of the resource's settings).

```ruby
Vmdb::Settings.encrypt_passwords!(resource.settings_for_resource.to_hash).to_yaml
```